### PR TITLE
CI: Kimi-K2.5 python launch and 4-layer CI test

### PR DESCRIPTION
**Pull request by Zhichenzzz**
## What

Adds a Python launch script for **Kimi-K2.5** full-parameter GRPO plus a CI smoke test, following the established model-script pattern (`run_qwen3_30b_a3b.py` / `run_glm5_744b_a40b.py` + the GLM-5 4-layer CI).

### Files
- **`scripts/run_kimi_k25.py`** — Python launch script with importable `prepare` / `execute` functions and `full-train` / `prepare` / `train` subcommands. Encodes the working config from `scripts/run-kimi-k25.sh`:
  - INT4 checkpoint feeds the SGLang rollout; Megatron loads a BF16 reference via the HF bridge (`--megatron-to-hf-mode bridge`) — no offline `torch_dist` conversion.
  - Dequantizes INT4→BF16 with `tools/convert_kimi_int4_to_bf16.py` in `prepare`.
  - MLA + MoE Megatron args reused from `scripts/models/kimi-k2-thinking.sh`; parallelism scales between single-node (4-layer) and the full 32-node setting.
- **`tests/e2e/megatron/test_kimi_k25_4layer_ci.py`** — single-node 8-GPU smoke test on the 4-layer pruned model `CharyZeng/Kimi-K2.5-4layer`. Registered with `register_cuda_ci(suite="stage-c-8-gpu-h100", labels=["model-scripts"])`, identical to the GLM-5 4-layer CI; auto-discovered by the suite runner and gated on the `run-ci-model-scripts` label.
- **`scripts/models/kimi-k25_4layer.sh`** — Megatron `MODEL_ARGS` override sourcing `kimi-k2-thinking.sh` and patching `num-layers=4`, `moe-layer-freq=[0,1,1,1]` (1 dense + 3 MoE).

### CI model
`CharyZeng/Kimi-K2.5-4layer` (public) is a 4-layer truncation of `moonshotai/Kimi-K2.5` — keeps embeddings / lm_head / final norm / full vision tower and decoder layers 0–3 — analogous to `Pinaster/GLM-5_4layer`.

## Test
Smoke test only verifies the training script is functional (not model accuracy). It runs the 4-layer model on one 8-GPU H100 node via the `model-scripts` CI suite.

### CODE REVIEW

File: scripts/run_kimi_k25.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+@dataclass
+class ScriptArgs(U.ExecuteTrainConfig):
+    mode: Literal["normal", "debug_minimal"] = "normal"
+    run_id: str = U.create_run_id()
+    model_org: str = "moonshotai"
+    model_name: str = "Kimi-K2.5"
+    megatron_model_type: str = "kimi-k2-thinking"
+    num_gpus_per_node: int = 8
+    enable_eval: bool = False
+    num_rollout: int = 20
```
Comment (yueming-yuan) [inline scripts/run_kimi_k25.py]: we usually set this to 3000 etc., or people might unexpectedly see their train ends at 20 rollout (

**Review by yueming-yuan — APPROVED** [review]
approved with tiny nit
